cmake/OpenCVCompilerOptions.cmake

```diff
@@ -40,16 +40,51 @@ if((CV_CLANG AND NOT CMAKE_GENERATOR MATCHES "Xcode")  # PCH has no support for
   set(ENABLE_PRECOMPILED_HEADERS OFF CACHE BOOL "" FORCE)
 endif()
 
+# add_extra_compiler_option(<option> [...]
+#                           [TURN_OFF_COMPILER_FLAG <flag> [<flag> ...]]
+#                           [TURN_OFF_LINKER_FLAG <flag> [<flag> ...]]
```

Comment:
I have made the renaming. Could explain what you think about when you are talking about patterns ?

Reply:
I mean to replace checks `if(... STREQUAL ...)` => `if(... MATHES "${regex_pattern}")`

In this case use `string(REGEX REPLACE ...)` instead of `string(REPLACE ...)`